### ISSUE TITLE
[RFC] Adding symfony/workflow 

### ISSUE THREAD

**Issue by ChangePlaces** [bolt/core#1357]
See https://github.com/bolt/bolt/issues/7851

tl;dr; merging in symfony/workflow concepts into here.

I'm thinking it'd be a top-level object like taxonomy / content types and would have its own config yaml file. Content types could assign a workflow to a content type, and a status field for keeping track of the current state could be created internally (with the option of being able to specify its name and the starting place). As several workflows can be assigned to a content type, this will need thought - e.g. the status field should be like `state_<workflowname>`.

It will also need its own field type for changing state on the form via the workflow's transitions.
